### DISCUSSION TITLE
[knitr] Frequently Asked Questions | /knitr/faq/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#945] [Comments]
https://yihui.org/knitr/faq/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
I'm afraid this is a little hard. I don't have time to help you for now, so I recommend you to ask questions on StackOverflow or TeX.StackExchange instead (with a minimal reproducible example and screenshots).

> Originally posted on 2017-07-19 05:04:38

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
I don't think PDF supports GIF. The only thing I'm aware of is the LaTeX package animate: <https://ctan.org/pkg/animate> It only works in Adobe Reader.

> Originally posted on 2017-08-16 22:13:54

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
You can call knitr from command line (via Rscript), so any editor/environment that supports calling external commands can call knitr, too. See some examples at <https://yihui.name/knitr/demo/editors/> I bet someone must have implemented an Atom plugin for knitr. I don't use Atom, so I'm not really sure.

> Originally posted on 2018-05-13 13:45:57

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks for the kind words! :)

> Originally posted on 2019-10-01 21:08:04

[a comment by an automated account is not shown]

**Comment by SAAD17296250** [comment]
Dear Sir,
How to resolve the following errors when compiling R Notebook into a document?  
All the required packages are successfully loaded, still the issue persists. Kindly help ASAP.

CODE:

```r
# pipe operator
x<-5:15
mean(x)
x%>%mean

sum(z,na.rm=TRUE)
z%>%sum(na.rm=TRUE)

sum(is.na(z))
z%>%is.na%>%sum

mean(is.na(z))
z%>%is.na%>%mean
```

OUTPUT:

```
|...................................                 |  67% (unnamed-chunk-1)

processing file: qwerty.Rmd
Quitting from lines 13-124 (qwerty.Rmd) 
Error in x %>% mean : could not find function "%>%"
Calls: <Anonymous> ... withVisible -> eval_with_user_handlers -> eval -> eval
                                                                                                            
Execution halted
```

CODE:

```r
data(diamonds)
View(diamonds)

# base histogram(s)
hist(x=diamonds$carat,main="FREQUENCY vs. CARAT",xlab="carat",ylab="frequency")

# base scatter plot(s)
(plot(x=diamonds$carat,y=diamonds$price,xlab="carat",ylab="price"))
```

OUTPUT:

```
 |...................................                 |  67% (unnamed-chunk-1)

processing file: qwerty.Rmd
                                                                                                            
Quitting from lines 13-24 (qwerty.Rmd) 
Error in as.data.frame(x) : object 'diamonds' not found
Calls: <Anonymous> ... eval_with_user_handlers -> eval -> eval -> View -> as.data.frame
Execution halted
```

**Comment by williamzhang-UOA** [comment]
感谢大神...
是从阿德莱德大学网站里 摸过来的
祝顺利

**Reply by yihui** [reply]
谢谢！你在奥克兰？我上周刚去了一趟奥克兰，前天刚回来。

**Comment by williamzhang-UOA** [comment]
阿德莱德是在澳大利亚 哈哈

On Tue, 17 Sept 2024 at 11:08, Yihui Xie ***@***.***> wrote:

> 谢谢！你在奥克兰？我上周刚去了一趟奥克兰，前天刚回来。
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/yihui/yihui.org/discussions/945#discussioncomment-10665379>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJIZURSTQTMSEQXFPX7R6C3ZW6BZ3AVCNFSM6AAAAABOKL7SUWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANRWGUZTOOI>
> .
> You are receiving this because you commented.Message ID: <yihui/yihui.
> ***@***.***>
>


-- 
Feng Zhang (William)
IT Graduate
M.ICT (University of Tasmania)
Adelaide. Australia
M:0466689999

**Reply by yihui** [reply]
嗯，我知道阿德莱德在澳大利亚，但你的用户名里有 UOA，所以我猜你可能是奥克兰大学毕业的。

**Comment by williamzhang-UOA** [comment]
haha UOA 就是 University of Adelaide 我才开始读 第二学期

On Tue, 17 Sept 2024 at 12:32, Yihui Xie ***@***.***> wrote:

> 嗯，我知道阿德莱德在澳大利亚，但你的用户名里有 UOA，所以我猜你可能是奥克兰大学毕业的。
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/yihui/yihui.org/discussions/945#discussioncomment-10665738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJIZURUOGYA4HICQYDBZABLZW6LTVAVCNFSM6AAAAABOKL7SUWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANRWGU3TGOA>
> .
> You are receiving this because you commented.Message ID: <yihui/yihui.
> ***@***.***>
>


-- 
Feng Zhang (William)
IT Graduate
M.ICT (University of Tasmania)
Adelaide. Australia
M:0466689999

**Reply by yihui** [reply]
嗦嘎。
